### PR TITLE
[W5.4][T14-A2] Xu Yiqing

### PR DESCRIPTION
**Pull request by eeching**
add command restore command

My original undo cannot pass the checks here and I cannot find why. So I rewrote it. Same function as shown during the lecture.

### CODE REVIEW

**Comment by eeching** [comment]
I have passed all JUnit tests and IO tests on my own laptop as the terminal indicates "Test result: PASS". However, the Travis CI showed that checks have failed. Can I know what should I do?

**Comment by OscarWang114** [comment]
Congrats on passing the test!

File: doc/UserGuide.md [hunk 1]
```diff
@@ -81,6 +81,20 @@ Examples:
...
+> Revoke the most recent deletion.
+
+Examples:
+* `list`<br>
+  `delete 2`<br>
+  Deletes the 2nd person in the address book.
+* `find Betsy`<br>
+  `restore`<br>
+  Restore the person that was deleted in 'delete 2'.
+
```
Comment (K1ang) [inline doc/UserGuide.md]: How about supporting `add` function too?

File: src/seedu/addressbook/storage/StorageForDeleted.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import java.io.Writer;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Stack;
+
+import static seedu.addressbook.commands.AddCommand.MESSAGE_DUPLICATE_PERSON;
+
+public class StorageForDeleted {
```
Comment (K1ang) [inline src/seedu/addressbook/storage/StorageForDeleted.java]: Do remember to add header comments to classes and nontrivial methods.

File: src/seedu/addressbook/storage/StorageForDeleted.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    private static Stack<ReadOnlyPerson> storageForDeleted;
+
+    public StorageForDeleted() {
+        this.storageForDeleted = new Stack<ReadOnlyPerson>();
+    }
+
+    public void storeTheDeleted(ReadOnlyPerson personBeingDeleted) throws UniquePersonList.DuplicatePersonException {
+        // System.out.println("i m at storeTheDeleted");
+        storageForDeleted.add(0, personBeingDeleted);
+        // System.out.println(storageForDeleted.peek().toString());
```
Comment (K1ang) [inline src/seedu/addressbook/storage/StorageForDeleted.java]: How about using `push()` instead

**Comment by K1ang** [comment]
Some comments added. Please close the PR after reading comments.

File: doc/UserGuide.md (hunk 1 above)
Comment (eeching) [reply doc/UserGuide.md]: Sure! The original one has the add function, yet I have trouble passing the check here. So I rewrote a simple one here. I will add back the add function during the recess week! May I know if I have passed this LO or  do I have to do one more enhancement? Thank you!

**Pull request closed without merging**